test/uk/gov/hmrc/helptosave/repo/MongoEmailStoreSpec.scala

```diff
@@ -40,111 +43,115 @@ class MongoEmailStoreSpec extends TestSupport with MongoTestSupport[EmailData, M
       .expects(input)
       .returning(output.fold[Try[String]](Failure(new Exception("uh oh")))(Success(_)))
 
-  def newMongoStore() = new MongoEmailStore(mockMongo, crypto, mockMetrics) {
+  def newMongoEmailStore(reactiveMongoComponent: ReactiveMongoComponent) =
+    new MongoEmailStore(reactiveMongoComponent, crypto, mockMetrics) {
 
-    override def indexes: Seq[Index] = {
-      // this line is to ensure scoverage picks up this line in MongoEnrolmentStore -
-      // we can't really test the indexes function, it doesn't affect the behaviour of
-      // the class only its performance
-      super.indexes
-      Seq.empty[Index]
     }
 
-    override def doUpdate(encryptedEmail: String, nino: NINO)(implicit ec: ExecutionContext): Future[Option[EmailData]] =
-      mockDBFunctions.update(EmailData(nino, encryptedEmail))
-
-    override def find(query: (String, Json.JsValueWrapper)*)(implicit ec: ExecutionContext): Future[List[EmailData]] =
-      query.toList match {
-        case head :: Nil ⇒ mockDBFunctions.get(head._2)
-        case _           ⇒ fail("Find function expected only 1 parameter")
-      }
-  }
-
   "The MongoEmailStore" when {
 
+    val nino = "NINO"
+    val email = "EMAIL"
+    val encryptedEmail = "ENCRYPTED"
+
     "updating emails" must {
 
-      val nino = "NINO"
-      val email = "EMAIL"
-      val encryptedEmail = "ENCRYPTED"
       val data = EmailData(nino, encryptedEmail)
 
-        def update(nino: NINO, email: String): Either[String, Unit] =
-          Await.result(mongoStore.storeConfirmedEmail(email, nino).value, 5.seconds)
+        def update(nino: NINO, email: String, emailStore: MongoEmailStore): Either[String, Unit] =
+          Await.result(emailStore.storeConfirmedEmail(email, nino).value, 5.seconds)
+
+        def getConfirmedEmail(nino: NINO, emailStore: MongoEmailStore): Either[String, Option[String]] =
+          Await.result(emailStore.getConfirmedEmail(nino).value, 5.seconds)
 
       "store the email in the mongo database" in {
-        inSequence {
-          mockEncrypt(email)(encryptedEmail)
-          mockUpdate(data)(Right(None))
+        withMongo { reactiveMongoComponent ⇒
+          val emailStore = newMongoEmailStore(reactiveMongoComponent)
+
+          inSequence {
+            mockEncrypt(email)(encryptedEmail)
+            mockDecrypt(encryptedEmail)(Some(email))
+          }
+
+          update(nino, email, emailStore)
+
+          val storedEmail = getConfirmedEmail(nino, emailStore)
+          storedEmail shouldBe Right(Some(email))
+
+          emailStore.collection.db.connection.askClose()
```

Comment:
do we need this here ? if so we its not there in other tests ?

Reply:
I've removed and the tests still passed, so its not needed